ls-ecaps.c

```diff
@@ -701,7 +701,7 @@ cap_dvsec_cxl(struct device *d, int id, int where)
   if (id != 0)
     return;
 
-  if (!config_fetch(d, where + PCI_CXL_CAP, 12))
+  if (!config_fetch(d, where + PCI_CXL_CAP, 0x38 - 0xa))
```

Comment:
If you are changing _size_ by its calculation then I would suggest to use new/existing macros instead of harcoding magic numbers like `0x38` or `0xa`. There are:
```c
#define PCI_CXL_CAP		0x0a	/* CXL Capability Register */
#define PCI_CXL_DEV_CAP			0x0a	/* CXL Capability Register */
#define PCI_CXL_DEV_RANGE2_BASE_LO	0x34
```
which seems to be source of those magic numbers.